File: Python/cell.py
class Cell(object):

	def __init__(self, size, lines):
		self.hints = [i + 1 for i in range(size)]
		self.lines = lines
		self._value = None

	@property
	def value(self):
		return self._value

	@value.setter
	def value(self, value):
		self._value = value
		del self.hints[:]
		for line in self.lines:
			line.removeHint(value)

	def removeHint(self, hint):
		if hint in self.hints:
			self.hints.remove(hint)
			return True
		return False

	def removeHints(self, hints):
		removed = False
		for hint in hints:
			removed = self.removeHint(hint) or removed
		return removed

File: Java/src/main/java/com/clovisleoncio/kenkensolver/Cell.java
package com.clovisleoncio.kenkensolver;

import java.util.HashSet;
import java.util.Set;

import lombok.Data;
import lombok.ToString;

@Data
@ToString(exclude = "lines")
public class Cell {
	
	public static final Cell cellForBoardSize(int boardSize) {
		Cell cell = new Cell();
		for (int i = 0; i < boardSize; i++) {
			cell.hints.add(i + 1);
		}
		return cell;
	}
	
	private Integer value;
	private Set<Integer> hints = new HashSet<>();
	private Line[] lines;
	
	public void setLines(Line... lines) {
		this.lines = lines;
	}
	
	public Line[] getLines() {
		return lines;
	}

	public void removeHint(Integer hint) {
		hints.remove(hint);
	}
	
}
